lib/Github/ResultIterator.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace Github;
+
+use Github\Api\ApiInterface;
+use Github\Api\Search;
+use Iterator;
+
+/**
+ * The result iterator class.
+ *
+ * This class allows you to iterate over a large data set with a low memory
+ * overhead due to pagination under the hood.
+ *
+ * @author Graham Campbell <graham@alt-three.com>
+ */
+class ResultIterator implements Iterator
+{
+    /**
+     * @var \Github\ResultPagerInterface
```

Comment:
sure - i'm just in habit of using FQCNs in phpdoc for readability - i'll change it when i have a moment